### ISSUE TITLE
Jenkins POD doesn't come up with jenkins-operator image created from sourcecode

### ISSUE THREAD

**Issue by pratham-m** [jenkinsci/kubernetes-operator#249]
Hi,

I have created an image of jenkins-operator using the `make container-runtime-build` target option in the Makefile, that is available with the source code. The image is build successfully.
I then deployed the jenkins-operator with steps mentioned below:
`kubectl -n default apply -f deploy/crds/jenkins_v1alpha2_jenkins_crd.yaml`
`kubectl -n default apply -f deploy/all-in-one-v1alpha2.yaml` Note: I updated this file to use image that was created from source code.
The jenkins-operator POD started.

After this, I tried to start jenkins instance as per step provided in [deploy jenkins](https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/deploy-jenkins/) webpage.
After creating jenkins custom resource using `kubectl -n default apply -f deploy/crds/jenkins_v1alpha2_jenkins_cr.yaml` the POD doesn't come up.

If I use the `virtuslab/jenkins-operator:v0.3.1` image to deploy the jenkins-operator instead of the image created using source code, the jenkins POD comes up.

Am i missing something? Any help would be useful. Thanks in advance.

Used `kubectl -n default get jenkins -o yaml` to compare both setups and found following differences:
Additions:
` spec:`
`    jenkinsAPISettings:`
`       authorizationStrategy: createUser`
`    serviceAccount: {}`
Missing part:
`  status:`
`    provisionStartTime: "2020-01-16T13:42:45Z"`
`    userAndPasswordHash: vj9TakfbQK0A9Aak9/Zbt6Jm51emJ/jLquShLOZI7VE=`

**Comment by tomaszsek** [comment]
Hi @pratham-m,

Please provide the logs of your jenkins operator.

Cheers

**Comment by pratham-m** [comment]
Here is the log of jenkins-operator container.
[jenkins-operator-container-logs.txt](https://github.com/jenkinsci/kubernetes-operator/files/4076497/jenkins-operator-container-logs.txt)

**Comment by pratham-m** [comment]
I am using `kubectl version`

`Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}`
`Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}`

**Comment by tomaszsek** [comment]
The operator doesn't have enough permissions. Make sure it has all permissions located here https://github.com/jenkinsci/kubernetes-operator/blob/master/deploy/role.yaml.

**Comment by pratham-m** [comment]
@tomaszsek thank you for the information. adding appropriate permissions in the YAML solved the issue. 

Now I am facing another issue: the jenkins instance pod comes up and then terminates immediately or in couple of seconds. Here is the operator log.
[operator-logs.txt](https://github.com/jenkinsci/kubernetes-operator/files/4086748/operator-logs.txt)

**Comment by tomaszsek** [comment]
```
Jenkins master pod restarted by operator: Jenkins pod node selector has changed, actual 'map[node-role.kubernetes.io/compute:true]' required 'map[]'	{"cr": "example"}
```
You have to add this `node-role.kubernetes.io/compute: true` to the CR under `spec.master.nodeSelector`. The Kubernetes added this itself.
I will add fix in the ` v0.3.2` version which will be released soon.

**Comment by tomaszsek** [comment]
The fix is on master branch.

**Comment by pratham-m** [comment]
Thanks for the fix. I will get the latest master and check.

**Comment by pawelprazak** [comment]
FYI https://github.com/jenkinsci/kubernetes-operator/releases/tag/v0.3.2 is avaiable now

**Comment by pratham-m** [comment]
@tomaszsek the `nodeSelector` change is working fine. thank you for the fix. 
@pawelprazak  thank you for the new release.

I am facing another issue as described below:

The Jenkins instance pod now comes up. the pod starts downloading HPI packages, howerev failure is observed while downloading some packages, as a result the jenkins-operators restarts the jenkins pod. I am using `jenkins/jenkins:lts` image to create the CR.

Here is the log for jenkins operator 
[operator-logs.txt](https://github.com/jenkinsci/kubernetes-operator/files/4103691/operator-logs.txt)
 and the jenkins instance 
[jenkins-example-logs.txt](https://github.com/jenkinsci/kubernetes-operator/files/4103706/jenkins-example-logs.txt)

I tried to download the failed HPI packages manually from browser, and that too gives NOT-Found error.

**Comment by tomaszsek** [comment]
I can download the https://updates.jenkins.io/2.204/latest/bouncycastle-api.hpi from the browser. There was some issue with the Jenkins plugin hosting.

**Comment by pratham-m** [comment]
Looks like some intermittent internet connectivity issue is causing CURL to throw error at my end. 
Closing this issue, since the `nodeSelector` bug is fixed in v0.3.2.
Thanks for the support.
